### PR TITLE
add `--package` to `azure-autorust`

### PR DESCRIPTION
**Pull request by ctaggart**
This allows the packages to generate to be limited. This is useful for development. Previously, `ONLY_SERVICES` was being hardcoded.

### CODE REVIEW

File: services/autorust/azure-autorust/src/main.rs [hunk 1]
```diff
@@ -1,9 +1,9 @@
 // cargo run --release -p azure-autorust
+// cargo run --release -p azure-autorust -- -p azure_svc_blobstorage
```
Comment (ctaggart) [inline services/autorust/azure-autorust/src/main.rs:2]: Example usage where only `azure_svc_blobstorage` will be generated.

File: services/autorust/azure-autorust/src/main.rs [hunk 2]
```diff
@@ -14,52 +14,63 @@ struct Args {
     /// Generate the publish GitHub workflows
     #[clap(long)]
     publish: bool,
+
+    /// Specify specific package to generate
+    #[clap(long = "package", short = 'p')]
```
Comment (ctaggart) [inline services/autorust/azure-autorust/src/main.rs:19]: `clap` is pretty nice

File: services/autorust/azure-autorust/src/main.rs [hunk 3]
```diff
@@ -14,52 +14,63 @@ struct Args {
     /// Generate the publish GitHub workflows
     #[clap(long)]
     publish: bool,
+
+    /// Specify specific package to generate
```
Comment (rylev) [inline services/autorust/azure-autorust/src/main.rs:18]: Nit: we probably should mention that this takes a list of packages.

File: services/autorust/azure-autorust/src/main.rs [hunk 4]
```diff
@@ -14,52 +14,63 @@ struct Args {
...
+    #[clap(long = "package", short = 'p')]
+    packages: Vec<String>,
+}
+impl Args {
```
Comment (rylev) [inline services/autorust/azure-autorust/src/main.rs:22]:
```suggestion
}

impl Args {
```

File: services/autorust/azure-autorust/src/main.rs [hunk 5]
```diff
@@ -14,52 +14,63 @@ struct Args {
...
+    pub fn packages(&self) -> Vec<&str> {
+        self.packages.iter().map(String::as_str).collect()
```
Comment (rylev) [inline services/autorust/azure-autorust/src/main.rs:24]: Just a thought: This seems somewhat wasteful - maybe returning `impl Iterator<Item = &str>` would work better?

**Review by rylev — APPROVED** [review]
Looks good!

File: services/autorust/azure-autorust/src/main.rs (hunk 3 above)
Comment (cataggar) [reply services/autorust/azure-autorust/src/main.rs]: I added to the help that `Multiple accepted.`
```
USAGE:
    azure-autorust.exe [OPTIONS]

OPTIONS:
    -h, --help                 Print help information
    -p, --package <PACKAGE>    Specify specific package to generate. Multiple accepted
        --publish              Generate the publish GitHub workflows     
```

**Pull request merged**